x/session/keeper/session_hydrator_test.go

```diff
@@ -0,0 +1,290 @@
+package keeper_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	keepertest "pocket/testutil/keeper"
+	"pocket/testutil/sample"
+	"pocket/x/session/keeper"
+	"pocket/x/session/types"
+)
+
+func TestSession_HydrateSession_Success_BaseCase(t *testing.T) {
+	sessionKeeper, ctx := keepertest.SessionKeeper(t)
+	blockHeight := int64(10)
+
+	sessionHydrator := keeper.NewSessionHydrator(keepertest.TestApp1Address, keepertest.TestServiceId1, blockHeight)
+	session, err := sessionKeeper.HydrateSession(ctx, sessionHydrator)
+	require.NoError(t, err)
+
+	// Check the header
+	sessionHeader := session.Header
+	require.Equal(t, keepertest.TestApp1Address, sessionHeader.ApplicationAddress)
+	require.Equal(t, keepertest.TestServiceId1, sessionHeader.ServiceId.Id)
+	require.Equal(t, "", sessionHeader.ServiceId.Name)
+	require.Equal(t, int64(8), sessionHeader.SessionStartBlockHeight)
```

Comment:
`sessionHeader.SessionStartBlockHeight` should be `9` assuming we start with `BlockHeight#1`
```
Blocks    [ 1 2 3 4 ][ 5 6 7 8 ][ 9 10 [
Sessions  [    1    ][    2    ][   3  [

Reply:
The first (genesis) block is 0 though ;)